src/WebJobs.Script.WebHost/App_Start/AutofacBootstrap.cs

```diff
@@ -30,7 +30,7 @@ internal static void Initialize(ScriptSettingsManager settingsManager, Container
             builder.Register<WebScriptHostManager>(ct => ct.Resolve<WebHostResolver>().GetWebScriptHostManager(settings)).ExternallyOwned();
             builder.Register<WebHookReceiverManager>(ct => ct.Resolve<WebHostResolver>().GetWebHookReceiverManager(settings)).ExternallyOwned();
             builder.Register<HostPerformanceManager>(ct => ct.Resolve<WebHostResolver>().GetPerformanceManager(settings)).ExternallyOwned();
-            builder.Register<ILoggerFactory>(ct => ct.ResolveOptional<WebScriptHostManager>()?.Instance.ScriptConfig.HostConfig.LoggerFactory).ExternallyOwned();
```

Comment:
This was the bug - the host Instance will be null in cases where it can't be created due to errors